cluster-autoscaler/cloudprovider/magnum/magnum_manager_heat_test.go

```diff
@@ -260,8 +260,8 @@ var stackGetKubeMinionsStackResponse = fmt.Sprintf(`
         "outputs":[
             {
                 "output_value":{
-                    "0":"ffbc651da661462d94352e01f3020688",
```

Comment:
That's the thing I still don't understand well. Where did you got this sample UUID when you worked on this initially? And why the instance ID in MachineID and ProviderID are using different format?

Reply:
And based on the changed you're doing in this test, I can't see how the case is covered. Did I miss something?

Reply:
My clusters are still using IPs as references in their stacks so I had to modify this example response and I was generating IDs with python `uuid.uuid4().hex` when I should have been using `str(uuid.uuid4())` to get the right format for openstack.

The delete tests use the example stack response and try to delete one of the minions.
https://github.com/kubernetes/autoscaler/blob/d887ba5e873e7fadfae1c7bae949cfab6d97e8db/cluster-autoscaler/cloudprovider/magnum/magnum_manager_heat_test.go#L155-L157
Now that I'm thinking about it, that ProviderID should have dashes in as well, but it isn't used anyway.

As for why the MachineID and ProviderID are different, it seems that the kubelet checks the file `/etc/machine-id` [by default](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/)
```
--machine-id-file string | Comma-separated  list of files to check for machine-id. Use the first one that exists.  (default "/etc/machine-id,/var/lib/dbus/machine-id")
```

```
[fedora@scaler-53-w74zzswdismh-minion-0 ~]$ cat /etc/machine-id
9f7da7a47b514e5197c37607828e850f
```
but I think the ProviderID would be set by the cloud provider by querying the cloud and so would use the openstack format.
```
Name                              MachineID                          ProviderID
scaler-53-w74zzswdismh-master-0   8faa9ed686c44e9ea30eb8fb7ca30b21   openstack:///8faa9ed6-86c4-4e9e-a30e-b8fb7ca30b21
scaler-53-w74zzswdismh-minion-0   9f7da7a47b514e5197c37607828e850f   openstack:///9f7da7a4-7b51-4e51-97c3-7607828e850f
```

Reply:
That's a very impressive clarification. Thank you so much, @tghartland